Update Getting Started (#4253)
circleci/circleci-docs: jekyll/_cci2/getting-started.md

> ## Prerequisites for Running Your First Build

Reviewer: Could you take out the `2.0` from the intro para above? as there's no other CircleCI?

{:.no_toc}

* Some basic knowledge of Git.
* A GitHub account, of which you are logged into.
* An account on CircleCI.
* Some basic terminal or `bash` knowledge and prior experience using the command line is helpful.

## Creating a Repository

Begin by creating a new repository on GitHub. You may skip this section if you intend to use an existing repository.

1. Navigate to GitHub and [create a new repository](https://github.com/new). 
1. Input the name of your repository, in this case "hello-world", then click

> `Initialize this repository with a README`. Finally, click `Create repository`.

Reviewer: I find the button clicks written as inline literals a bit confusing to look at. I think we normally just capitalise button labels? or put in " " or bold them? I'm happy with any of them as long as we're consistent 🤷‍♀ 😄

Author: Good point - I think we use bold elsewhere, so I'll standardize to that.

![Creating a Repository]( {{ site.baseurl }}/assets/img/docs/getting-started--new-repo.png){:.img--bordered}

## Setting up CircleCI

If you have not yet, create an account on CircleCI by navigating to [the signup
page](https://circleci.com/signup/) and clicking on `Sign Up with GitHub`.

1. Navigate to the CircleCI [Project Page](https://app.circleci.com/projects/).
1. If you are part of any organizations, you will need to select the

>    organization you are part in order to setup your repository with CircleCI.

Reviewer: ```suggestion
   organization you wish to work under in order to setup your repository with CircleCI.
```

Reviewer: just an idea, think there's a word missing and couldn't guess which one!

  World` configuration example at the bottom of the list. 

    ![Getting a sample configuration]( {{ site.baseurl }}/assets/img/docs/getting-started--sample-config.png){:.img--bordered}
  
    **Note:** Based on which language you choose you can view related 
    documentation in the sidebar on the right of the screen 

1. Click `Start Building` and follow the prompt to add your config to a new
   branch and start your first pipeline. **Note:** You can also manually add a
   `config.yml` to your project root under a `.circleci` folder and push this to
   your VCS provider. 

    ![Automatic Commit]( {{ site.baseurl }}/assets/img/docs/getting-started--auto-commit.png){:.img--bordered}

1. You will be taken to the Pipelines page where you can view your newly running
   build. 

## Digging Into Your First Pipeline

You should see your pipeline start to run automatically—and pass! So, what just
happened? Click on the green `Success` button on your pipeline to investigate
the following parts of the run: 

![First Successful Pipeline]( {{ site.baseurl }}/assets/img/docs/getting-started--first-success.png){:.img--bordered}

> 1. **What workflows ran?**: After clicking "Success", we are taken to a page

Reviewer: Which workflows ran? maybe?

   listing the jobs that ran. If this is your first build, you probably only ran
   **one job**  (which automatically runs inside **one workflow**).  In our

>    case, we only ran one job, called `Build`. Click on `Build` and let's

Reviewer: In the image the job is `welcome/run`, rather than `build`

Author: fixed!

   investigate the steps of our job. 

>   ![Investigate build]( {{ site.baseurl }}/assets/img/docs/getting-started--first-success-workflow.png){:.img--bordered}

Reviewer: Needs two tab indent for numbered steps to flow

Author: Fixed! Using two tab indents caused it to become a `pre` block, but 3 spaces worked 🤷‍♂ !

1. **Spin up environment:** CircleCI used an [orb](https://circleci.com/orbs) to
   help provide some defaults for this project. By using an orb, we can get
   quick access to common configuration. In this case,
   `circleci/welcome-orb@0.4.1` provides a "pre-built" job you can run which
   simply greets the user. 

1. **Views step results:** Every job is made up of a series of steps - some
   steps, like
   [`checkout`]({{site.baseurl}}/2.0/configuration-reference/#checkout) are

>    built-in commands in CircleCI. Other steps are specified by a user to achieve

Reviewer: In the config ref `checkout` and others that do magic fancy stuff are referred to as 'special steps', maybe that rather that build-in? I don't know...

### Breaking Your Build!
{:.no_toc}

Let's get a bit more complex. Let's edit our `.circleci/config.yml` file now. On
GitHub, it is possible to edit files directly. Use the URL below and substitute
the name of your repository and username (replace the text with`{backets}`) and
then paste it in your browser. If you are already familiar with Git, use your
text-editor and push your changes with git. 

`https://github.com/{username}/{repo}/edit/circleci-project-setup/.circleci/config.yml`

Let's use the [Node orb](https://circleci.com/orbs/registry/orb/circleci/node). Paste the following into your `config.yml`

```yaml
version: 2.1
orbs:
  node: circleci/node@1.1
jobs:
  build:
    executor:
      name: node/default
      tag: '10.4'
    steps:
      - checkout
      - node/with-cache:
          steps:
            - run: npm install
      - run: npm run test
```

Then, commit your change in the GitHub editor and return to the Projects page
in CircleCI. You should see a new pipelines running... and it will fail! What's going on?

The Node orb runs some common Node tasks. Because we are working with an empty
repository, running `npm run test`, a Node script, causes our configuration to

> fail.  How would we fix this? You would need to setup a Node project in your

Reviewer: ```suggestion
fail.  How would we fix this? You would need to set up a Node project in your
```

repository;  a topic for another tutorial. You can view several [demo
applications]({{site.baseurl}}/2.0/demo-apps/) that go into more detail on
setting up CircleCI with various languages and frameworks. 

## Using the Workflows Functionality
{:.no_toc}

> You do not have to use Orbs to use CircleCI. The following example details how

Reviewer: ```suggestion
You do not have to use orbs to use CircleCI. The following example details how
```

to create a custom configuration that also uses the [workflow
feature]({{site.baseurl}}/2.0/workflows) of CircleCI. 

1. Take a moment and read the comments in the code block below. Of course, we

> don't want to be copying and pasting code without understanding what we're

Reviewer: ```suggestion
do not want to be copying and pasting code without understanding what we are
```

      one: # This is our first job.
        docker: # it uses the docker executor
          - image: circleci/ruby:2.4.1 # specifically, a docker image with ruby 2.4.1
        # Steps are a list of commands to run inside the docker container above.
        steps:
          - checkout # this pulls pulls code down from GitHub
          - run: echo "A first hello" # This prints "A first hello" to stdout.
          - run: sleep 25 # a command telling the job to "sleep" for 25 seconds.
      two: # This is our second job.
        docker: # it runs inside a docker image, the same as above.
          - image: circleci/ruby:2.4.1
        steps:
          - checkout
          - run: echo "A more familiar hi" # We run a similar echo command to above.
          - run: sleep 15 # and then sleep for 15 seconds.
    # Under the workflows: map, we can coordinate our two jobs, defined above.
    workflows:
      version: 2
      one_and_two: # this is the name of our workflow
        jobs: # and here we list the jobs we are going to run.
          - one
          - two
    ```

1. Commit these changes to your repository and navigate back over to the

>     CircleCI. You should see your CircleCI pipeline running.

Reviewer: CircleCI Pipelines page maybe?

> ### SSH into Your {% comment %} TODO: Job {% endcomment %}Build

Reviewer: woah I didn't know it was possible to comment out like that!

Author: Haha yeah, I didn't write it, so I learned that recently too :)
